src/printer/embed.ts

```diff
@@ -34,37 +34,53 @@ export function embed(
 		const textContent = printRaw(jsxNode);
 
 		let content: Doc;
-		content = textToDoc(textContent, {
-			...opts,
-			parser: expressionParser,
-		});
 
-		content = stripTrailingHardline(content);
+		try {
+			content = textToDoc(textContent, {
+				...opts,
+				parser: '__js_expression',
+			});
+		} catch (e) {
+			// The `__js_expression` parser should be able to handle 99% of supported expressions, however there's some very rare cases
+			// where we unfortunately need the full `babel-ts` parser. This happens notably with comments. In the future, what
+			// we should probably attempt is to make the expression compatible with `__js_expression` through our `makeNodeJSXCompatible` function
+			content = wrapParserTryCatch(textToDoc, textContent, {
+				...opts,
+				parser: 'babel-ts',
+			});
+
+			content = stripTrailingHardline(content);
+
+			// HACK: We can't strip the trailing hardline if an expression starts with an inline comment or the ending curly
+			// bracket will end on the same line as the comment, which breaks the expression
+			if (textContent.trimStart().startsWith('//') && /\n/.test(textContent)) {
+				return group(['{', indent([hardline, content]), hardline, lineSuffixBoundary, '}']);
+			}
+		}
 
 		// Create a Doc without the things we had to add to make the expression compatible with Babel
 		const astroDoc = mapDoc(content, (doc) => {
-			if (typeof doc === 'string' && doc.startsWith('__PRETTIER_SNIP__')) {
```

Comment:
We can't use this anymore when using `__js_expression` because the full content we want to remove doesn't end up in a single nice string